### PR TITLE
Add teams sso integration

### PR DESCRIPTION
**Pull request by asnow003**

### CODE REVIEW

File: src/services/sso.auth.service.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  expect(authService.isSSO).toBeTruthy();
+});
+
+/*
+it("can get token", done => {
+  const tokenValue = "faketoken";
+
+  const authService = new SSOAuthService();
+  microsoftTeams.authentication.getAuthToken.mockImplementationOnce({
+
```
Comment (asnow003) [inline src/services/sso.auth.service.test.js]: I have to figure out how to mock these values :-)

File: manifest/sso/manifest.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  "permissions": [
+    "messageTeamMembers",
+    "identity"
+  ],
+  "validDomains": [
+    "taskmeow.com"
+  ],
+  "webApplicationInfo": {
+    "id": "36b1586d-b1da-45d2-9b32-899c3757b6f8",
+    "resource": "api://???/???"
```
Comment (asnow003) [inline manifest/sso/manifest.json]: Need the resource value

Comment (ydogandjiev) [reply manifest/sso/manifest.json]: Use "api://taskmeow.com/36b1586d-b1da-45d2-9b32-899c3757b6f8"

Comment (ydogandjiev) [reply manifest/sso/manifest.json]: We will need to make a change in Teamspace-Web to handle the domain extraction from the resource URI

File: src/services/sso.auth.service.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    // Initialize the Teams SDK
+    microsoftTeams.initialize();
+
+    // Flag that this service assumes using Teams SSO
+    this.isSSO = true;
+  }
+
+  getUser() {
+    return new Promise((resolve, reject) => {
+      microsoftTeams.authentication.getUser({
```
Comment (ydogandjiev) [inline src/services/sso.auth.service.js]: Hmm... This part will always fail :(

Comment (ydogandjiev) [reply src/services/sso.auth.service.js]: I wonder if there is a way to decode the auth token and get this information out of it...

Comment (asnow003) [reply src/services/sso.auth.service.js]: Oh I can do that...I did that somewhere else.

**Pull request merged**